packages/components/Button/src/FAB/FABCore.tsx

```diff
@@ -64,21 +69,49 @@ export const FAB = compose<FABType>({
       }
       const label = accessibilityLabel ?? childText;
 
-      const fabWithoutShadow = (
-        <Slots.root {...mergedProps} accessibilityLabel={label}>
+      const buttonContent = (
+        <React.Fragment>
           {icon && <Slots.icon {...iconProps} />}
           {showContent &&
             React.Children.map(children, (child) =>
               typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
             )}
+        </React.Fragment>
+      );
+      const buttonContentWithRoot = (
+        <Slots.root {...mergedProps} accessibilityLabel={label}>
+          {buttonContent}
         </Slots.root>
       );
 
       const hasShadow = Platform.OS === 'ios';
+      const hasRipple = Platform.OS === 'android';
+
       if (hasShadow) {
-        return <Slots.shadow>{fabWithoutShadow}</Slots.shadow>;
+        return <Slots.shadow>{buttonContentWithRoot}</Slots.shadow>;
+      } else if (hasRipple) {
+        const [outerStyleProps, innerStyleProps] = extractOuterStylePropsAndroid(mergedProps.style);
+        return (
+          <Slots.rippleContainer style={outerStyleProps}>
+            <Slots.root
+              accessibilityLabel={label}
+              {...mergedProps}
+              style={innerStyleProps}
+              onPressIn={memoize(() => {
+                setRippleStateAndroid(true);
+                mergedProps.onPressIn?.();
+              })}
+              onPressOut={memoize(() => {
+                setRippleStateAndroid(false);
+                mergedProps.onPressOut?.();
+              })}
```

Comment:
Can we extract this logic to a hook, similar to what we do with `useOnPressWithFocus()` ?

Reply:
Actually, could this setting of `onPressIn/Out` be put in `useButton.ts` ?

Reply:
Had initially placed it in useButton.ts, but it was unnecessarily being also used for ToggleButton and CompoundButton.
Have removed this, shouldn't be required now that we have migrated to RN Pressable.